skins/laika/src/components/pages/donation_form/subpages/AddressPage.vue

```diff
@@ -91,10 +91,13 @@ export default Vue.extend( {
 		submit() {
 			( this.$refs.address as any ).validateForm().then( () => {
 				if ( this.$store.getters[ NS_ADDRESS + '/requiredFieldsAreValid' ] ) {
+					if ( this.$store.getters[ NS_PAYMENT + '/isDirectDebitPayment' ] &&
+						!this.$store.getters[ NS_BANKDATA + '/bankDataIsValid' ] ) {
+						return;
+					}
```

Comment:
👍 
Now that I see this code, it makes total sense for me to conditionally check in the component instead of complicating the store code with conditional and cross-namespace checks. Especially since the `NS_PAYMENT` namespace might become `NS_DONATION_PAYMENT` while NS_BANKDATA will be used both for donation and membership.